host/sgx/linux/vdso.c

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include "../vdso.h"
+#include <openenclave/host.h>
+#include <openenclave/internal/elf.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/trace.h>
+#include <sys/auxv.h>
+#include "../asmdefs.h"
+#include "../enclave.h"
+#include "../exception.h"
+#include "sgx.h" // Linux kernel header
+
+extern bool oe_is_avx_enabled;
+
+static elf64_sym_t _sgx_enter_enclave_sym;
+static vdso_sgx_enter_enclave_t _vdso_sgx_enter_enclave;
+
+oe_result_t oe_sgx_initialize_vdso(void)
```

Comment:
Better to use oe_once to guard calls to this function.

Reply:
This function is called by _initialize_enclave_host_impl, which is guarded by oe_once. Do you think we should do it here?

https://github.com/openenclave/openenclave/blob/b2db393080ddb25e1462bce1bb27ee2fe470c9be/host/sgx/create.c#L97